### ISSUE TITLE
ADDA on miniDomainNet

### ISSUE THREAD

**Issue by Franklin-Yao** [KaiyangZhou/Dassl.pytorch#9]
Hi,
I tried ADDA on miniDomainNet, but the acc on the target test set decrease from 0.29 to 0.01. Previously I tried it on office31 and it works fine. Do you know what is wrong?
I used the following scripts:
```
python tools/train.py
--root $DATA \
--trainer SourceOnly \
--source-domains real \ 
--target-domains sketch \
--dataset-config-file configs/datasets/da/mini_domainnet.yaml \
--config-file configs/trainers/da/source_only/mini_domainnet.yaml \
--output-dir output/source_only_minidomainnet
```
```
python train.py --trainer ADDA --source-domains real --target-domain sketch \
--dataset-config-file configs/datasets/da/mini_domainnet.yaml \
--config-file configs/trainers/da/source_only/mini_domainnet.yaml \
--output-dir output/adda_minidomainnet \
--init-weights output/source_only_minidomainnet/model/model.pth.tar-60
```

**Comment by KaiyangZhou** [comment]
ADDA's updating rule makes the network vulnerable to large datasets (e.g. >100 classes)

this is because aligning marginal distribution doesn't guarantee class-conditioned distribution to be aligned

**Comment by Franklin-Yao** [comment]
Thx.
